contracts/upgradeable_contracts/erc20_to_native/HomeBridgeErcToNative.sol

```diff
@@ -0,0 +1,90 @@
+import "../BasicHomeBridge.sol";
+import "../ERC677Bridge.sol";
+
+
+contract HomeBridgeErcToNative is EternalStorage, BasicBridge, BasicHomeBridge {
+
+    function () public payable {
+        require(msg.value > 0);
+        require(msg.data.length == 0);
+        require(withinLimit(msg.value));
+        address blockRewardAddress = blockRewardContract();
+        if(blockRewardAddress.call(abi.encodeWithSignature("mintedTotallyByBridge(address)",  address(this)))) {
+            IBlockReward blockReward = IBlockReward(blockRewardAddress);
+            uint256 totalMinted = blockReward.mintedTotallyByBridge(address(this));
+            require(msg.value <= totalMinted.sub(totalBurntCoins()));
+        }
+        setTotalSpentPerDay(getCurrentDay(), totalSpentPerDay(getCurrentDay()).add(msg.value));
+        setTotalBurntCoins(totalBurntCoins().add(msg.value));
+        address(0).transfer(msg.value);
+        emit UserRequestForSignature(msg.sender, msg.value);
+    }
+
+    function initialize (
+        address _validatorContract,
+        uint256 _dailyLimit,
+        uint256 _maxPerTx,
+        uint256 _minPerTx,
+        uint256 _homeGasPrice,
+        uint256 _requiredBlockConfirmations,
+        address _blockReward
+
+    ) public returns(bool)
+    {
+        require(!isInitialized());
+        require(_validatorContract != address(0));
+        require(_requiredBlockConfirmations > 0);
+        require(_minPerTx > 0 && _maxPerTx > _minPerTx && _dailyLimit > _maxPerTx);
+        require(_blockReward == address(0) || isContract(_blockReward));
+        addressStorage[keccak256(abi.encodePacked("validatorContract"))] = _validatorContract;
+        uintStorage[keccak256(abi.encodePacked("deployedAtBlock"))] = block.number;
+        uintStorage[keccak256(abi.encodePacked("dailyLimit"))] = _dailyLimit;
+        uintStorage[keccak256(abi.encodePacked("maxPerTx"))] = _maxPerTx;
+        uintStorage[keccak256(abi.encodePacked("minPerTx"))] = _minPerTx;
+        uintStorage[keccak256(abi.encodePacked("gasPrice"))] = _homeGasPrice;
+        uintStorage[keccak256(abi.encodePacked("requiredBlockConfirmations"))] = _requiredBlockConfirmations;
+        addressStorage[keccak256(abi.encodePacked("blockRewardContract"))] = _blockReward;
+        setInitialize(true);
+
+        return isInitialized();
+    }
+
+    function getBridgeMode() public pure returns(bytes4 _data) {
+        return bytes4(keccak256(abi.encodePacked("erc-to-native-core")));
+    }
+
+    function blockRewardContract() public view returns(IBlockReward) {
+        return IBlockReward(addressStorage[keccak256(abi.encodePacked("blockRewardContract"))]);
+    }
+
+    function totalBurntCoins() public view returns(uint256) {
+        return uintStorage[keccak256(abi.encodePacked("totalBurntCoins"))];
+    }
+
+    function setBlockRewardContract(address _blockReward) public onlyOwner {
+        require(_blockReward != address(0) && isContract(_blockReward));
```

Comment:
As per the security audit (https://hackmd.io/s/Sk6P7Uqtm) we received a comment that this check is not enough. Please consider to extend the check with `IBlockReward(_blockReward).bridgeContract() == this`.

Reply:
Yes, I think we can use `bridgesAllowedLength()` to cover the security recommendation.

Reply:
Super! Could you rework the current implementation? Thanks!

Reply:
Done